hardware/smart_utils.py

```diff
@@ -69,9 +79,8 @@ def read_smart_scsi(hwlst, device, optional_flag="", mode=""):
     vendor = ""
     product = ""
     for line in sdparm_cmd.stdout:
-        line = line.strip()
-        if isinstance(line, six.binary_type):
-            line = line.decode()
```

Comment:
You can do it much simpler: just pass `errors="ignore"` to this `decode` call:
```
>>> 'превед!'.decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 0: ordinal not in range(128)
>>> 'превед!'.decode(errors='ignore')
u'!'
```

Reply:
You are correct `decode(errors='ignore')` is the correct solution after re-testing. Perhaps I mixed up my results from attempting with `errors='replace'`.